### ISSUE TITLE
GSPO example is not working

### ISSUE THREAD

**Issue by mshaocong** [huggingface/trl#5298]
I ran the `gspo.py` example given in [https://github.com/huggingface/trl/blob/main/examples/scripts/gspo.py](https://github.com/huggingface/trl/blob/main/examples/scripts/gspo.py). The format reward is increasing well but the accuracy reward is nearly zero. Any suggestions?  

<img width="1629" height="398" alt="Image" src="https://github.com/user-attachments/assets/47d85197-c298-4b77-8973-8b801ddba371" />

**Comment by casinca** [comment]
Hey, it's not in the script but for reasoning, it's preferred to use `reasoning_accuracy_reward` (it's only mentioned in the TRL README)
So you could try changing to:

```diff

@@ -67,7 +67,7 @@ from trl import (
     get_quantization_config,
 )
-from trl.rewards import accuracy_reward, think_format_reward
+from trl.rewards import reasoning_accuracy_reward, think_format_reward
 
 
 # Enable logging in a Hugging Face Space
@@ -123,7 +123,7 @@ if __name__ == "__main__":
     trainer = GRPOTrainer(
         model=model_args.model_name_or_path,
         args=training_args,
-        reward_funcs=[think_format_reward, accuracy_reward],
+        reward_funcs=[think_format_reward, reasoning_accuracy_reward],
         train_dataset=train_dataset,
         eval_dataset=eval_dataset,
         peft_config=get_peft_config(model_args),
     )

```

&nbsp;

Also I feel like this part of the `SYSTEM_PROMPT` is a bit confusing:
```
        "The reasoning process and answer are enclosed within <think></think> tags, i.e., <think>\nThis is my "
        "reasoning.\n</think>\nThis is my answer."
```

Not sure if you see it the same way, it's like asking to have the answer inside think tag but the ie example shows it outside.

You might want to customize a bit the prompt, to force easier parsing. For example, I know Qwen3 models are good already with requesting `\boxed{answer}` type of format, so it shouldn't be too much steering or something else.

Hope one or both methods help

**Comment by mshaocong** [comment]
> Hey, it's not in the script but for reasoning, it's preferred to use `reasoning_accuracy_reward` (it's only mentioned in the TRL README) So you could try changing to:
> 
> @@ -67,7 +67,7 @@ from trl import (
>      get_quantization_config,
>  )
> -from trl.rewards import accuracy_reward, think_format_reward
> +from trl.rewards import reasoning_accuracy_reward, think_format_reward
>  
>  
>  # Enable logging in a Hugging Face Space
> @@ -123,7 +123,7 @@ if __name__ == "__main__":
>      trainer = GRPOTrainer(
>          model=model_args.model_name_or_path,
>          args=training_args,
> -        reward_funcs=[think_format_reward, accuracy_reward],
> +        reward_funcs=[think_format_reward, reasoning_accuracy_reward],
>          train_dataset=train_dataset,
>          eval_dataset=eval_dataset,
>          peft_config=get_peft_config(model_args),
>      )
>  
> 
> Also I feel like this part of the `SYSTEM_PROMPT` is a bit confusing:
> 
> ```
>         "The reasoning process and answer are enclosed within <think></think> tags, i.e., <think>\nThis is my "
>         "reasoning.\n</think>\nThis is my answer."
> ```
> 
> Not sure if you see it the same way, it's like asking to have the answer inside think tag but the ie example shows it outside.
> 
> You might want to customize a bit the prompt, to force easier parsing. For example, I know Qwen3 models are good already with requesting `\boxed{answer}` type of format, so it shouldn't be too much steering or something else.
> 
> Hope one or both methods help

Many thanks for your respnse! My problem has been fully addressed. I made two modifications: One is replacing it with the reasoning accuracy reward (and it doesn't work actually --- red curve), another is increasing the model size from 0.5B to 3B (green curve). 

<img width="1401" height="418" alt="Image" src="https://github.com/user-attachments/assets/3a434743-84be-4684-ab89-c7185b943867" />
